go-controller/cmd/ovnkube-trace/ovnkube-trace.go

```diff
@@ -498,14 +516,22 @@ func getPodInfo(coreclient *corev1client.CoreV1Client, restconfig *rest.Config,
 	}
 
 	// Find rtos MAC (this is the pod's first hop router).
-	podInfo.RtosMAC, err = getRouterPortMacAddress(coreclient, restconfig, podInfo, ovnNamespace, types.RouterToSwitchPrefix)
+	lrpPrefix := types.RouterToSwitchPrefix
+	if podInfo.TopologyType() == types.Layer2Topology {
+		lrpPrefix = types.GWRouterToJoinSwitchPrefix + types.GWRouterPrefix
```

Comment:
would you point me to where we create the LRP ? This combination of prefixes seems rather fishy. 

IIUC, it should be different: https://github.com/ovn-org/ovn-kubernetes/blob/86cdf533f9b049e415fbf2f14756b602c156a1bf/go-controller/pkg/ovn/gateway.go#L335